arch/arm/src/armv7-a/arm_vectors.S

```diff
@@ -169,6 +183,13 @@ arm_vectorirq:
 	/* Get the values for r15(pc) and CPSR in r3 and r4 */
 
 	ldr		r0, .Lirqtmp			/* Points to temp storage */
+
+#ifdef CONFIG_SMP
+	cpuindex r1						/* r1 = cpuindex */
```

Comment:
Is it possible directly get cpuindex?
mrc             CP15_MPIDR(r1)                  /* Get R1=MPIDR */
and             r1, r1, #MPIDR_CPUID_MASK
mov             r1, r1, lsl #MPIDR_CPUID_SHIFT  /* r1 = cpuindex */

Reply:
@Donny9 

Why do you want not to use the macro?